hikari/impl/event_manager.py

```diff
@@ -241,16 +241,17 @@ async def on_guild_create(self, shard: gateway_shard.GatewayShard, payload: data
                 for voice_state in voice_states.values():
                     self._cache.set_voice_state(voice_state)
 
-        recv_chunks = self._enabled_for_event(shard_events.MemberChunkEvent) or self._cache_enabled_for(
```

Comment:
you know if nobody's listening to chunk events and we're not caching for chunk events its just a ton of wasted bandwidth and time so not sure why that was removed

Reply:
It was moved bellow. All the other checks are done first before the enabled for event, since its the most expensive

Reply:
You're gonna have to dupe the enabled for event check for member chunk between here and the filtered decorator call so that seems like an arbitrary micro-optimisation